contracts/peripheral/YieldDirectorV2.sol

```diff
@@ -0,0 +1,491 @@
+        @param recipient_ Address of user receiving donated yield
+     */
+    function totalRedeemableBalance(address recipient_) public view returns (uint256) {
+        uint256[] storage receiptIds = recipientIds[recipient_];
+
+        uint256 agnosticRedeemable = 0;
+
+        for (uint256 index = 0; index < receiptIds.length; ++index) {
+            agnosticRedeemable += redeemableBalance(receiptIds[index]);
+        }
+
+        return agnosticRedeemable;
+    }
+
+    /**
+        @notice Redeem recipient's donated amount of sOHM at current index from one donor as gOHM
+        @param depositId_ Deposit ID for this donation
+    */
+    function redeemYield(uint256 depositId_) external {
+        uint256 amountRedeemed = _redeem(depositId_);
+
+        IERC20(gOHM).safeTransfer(msg.sender, amountRedeemed);
+    }
+
+    /**
+        @notice Redeem recipient's donated amount of sOHM at current index
+        @param depositId_ Deposit id for this donation
+    */
+    function redeemYieldAsSohm(uint256 depositId_) external {
+        uint256 amountRedeemed = _redeem(depositId_);
+
+        IERC20(sOHM).approve(address(staking), amountRedeemed);
+        staking.unwrap(msg.sender, amountRedeemed);
+    }
+
+    /**
+        @notice Redeem recipient's full donated amount of sOHM at current index as gOHM
+    */
+    function redeemAllYield() external {
+        uint256 amountRedeemed = _redeemAll();
+
+        IERC20(gOHM).safeTransfer(msg.sender, amountRedeemed);
+    }
+
+    /**
+        @notice Redeem recipient's full donated amount of sOHM at current index as gOHM
+    */
+    function redeemAllYieldAsSohm() external {
+        uint256 amountRedeemed = _redeemAll();
+        
+        IERC20(sOHM).approve(address(staking), amountRedeemed);
+        staking.unwrap(msg.sender, amountRedeemed);
+    }
+
+    /************************
+     * Internal Functions
+     ************************/
+
+    /**
+        @notice Creates a new deposit directing the yield from the deposited gOHM amount
+                to the prescribed recipient
+        @param amount_ Quantity of gOHM deposited redirecting yield to the recipient
+        @param recipient_ The address of the user who will be entitled to claim the donated yield
+    */
+    function _createDeposit(uint256 amount_, address recipient_) internal returns (uint256 depositId) {
+        depositId = _deposit(msg.sender, amount_);
+        recipientIds[recipient_].push(depositId);
+        recipientLookup[depositId] = recipient_;
+
+        emit Deposited(msg.sender, recipient_, amount_);
+    }
+
+    /**
+        @notice Increases the amount of gOHM directing yield to a recipient
+        @param depositId_ The global ID number of the deposit to add the additional deposit to
+        @param amount_ Quantity of new gOHM deposited redirecting yield to the current deposit's recipient
+    */
+    function _increaseDeposit(uint256 depositId_, uint256 amount_) internal {
```

Comment:
is this function really necessary if its just these 2 lines?

Reply:
not really? I could be convinced either way. on the one hand it makes it so that function implementations seem consistent, but also declaring the function means it doesn't actually save any lines of code

Reply:
On second look, I think its ok, to stay consistent with the rest of the internal functions.